opennmt/utils/hooks.py

```diff
@@ -182,3 +183,33 @@ def end(self, session):
     tf.logging.info("Evaluation predictions saved to %s", self._output_path)
     if self._post_evaluation_fn is not None:
       self._post_evaluation_fn(self._current_step, self._output_path)
+
+class LoadCheckpointHook(tf.train.SessionRunHook):
+  """"Hook that loads model variables from checkpoint before starting the training."""
+
+  def __init__(self, checkpoint_path):
+    self.checkpoint_path = checkpoint_path
+
+  def begin(self):
+    var_list = tf.train.list_variables(self.checkpoint_path)
+    self.values = {}
+    for name, shape in var_list:
+      if not name.startswith("optim"):
+        if not name.startswith("global_step") and not name.startswith("words_per_sec"):
+          self.values[name] = np.zeros(shape)
```

Comment:
This value is not very useful. Maybe you can change this first loop to only extract the variable names we care about, e.g.:

```python
names = []

# Fill the array with variable names...

for name in names:
  # Get the value.
```